CDPT-71 Add request method options to SAR creation and display

Polsonby opened the pull request:
## Description
When a London Disclosure user creates a SAR, they should be able to specify how the SAR was received by MOJ

## Self-review checklist
<!-- Action these things before requesting reviews -->
* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed


### Screenshots
![image](https://user-images.githubusercontent.com/1161161/176559060-6231af99-43d0-4e76-85d7-feb80b0fdc4a.png)

![image](https://user-images.githubusercontent.com/1161161/176559145-52396921-ba60-436d-af46-0a87721f277e.png)

![image](https://user-images.githubusercontent.com/1161161/176559319-b073c1a5-4cd3-4326-9086-c474cc36c409.png)

![image](https://user-images.githubusercontent.com/1161161/176559348-1c6d46e2-123b-4f09-a708-f424780b8708.png)

### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CDPT-71
https://dsdmoj.atlassian.net/browse/CDPT-89

### Deployment
This PR contains a data migration - see the Readme. When it is deployed onto a namespace, run `rake data:migrate` to populate the SAR case request_method values.

### Manual testing instructions
Sign into the app as David Attenborough and create a new SAR case. You should be able to specify the request method from the options Post, Email, Web Portal and so on. 

When the case details page is displayed the correct request method should be displayed. 

When an Admin user creates a SAR case, the default settings should create a valid case using "Post" as the request method.

Polsonby committed:
Add request method to SAR JSON properties with basic validation

Polsonby committed:
Create a SAR specifying request method

Polsonby committed:
Fix validation messages on the new SAR case form to reflect human label

Polsonby committed:
Admin case creation for SARs specifies request method

Polsonby committed:
Display SAR case with request method

Polsonby committed:
Remove redundant reference to SAR from labels

Polsonby committed:
Fix tests for SAR Internal Review which inherits from SAR Standard

Polsonby committed:
Set a default request method for SAR IR cases

Because this correspondence type inherits from SAR Standard, but the request method is not required for this case type, we set it to a neutral default value

zheileman reviewed (approved):
LGTM 👍 
I've added a few suggestions but approving anyways as these are not critical 🥳

Polsonby committed:
Refactor to remove callback on SAR IR and apply conditional validation instead

Polsonby committed:
Remove accidental focused test

Polsonby committed:
Restore accidentally commented out line

Polsonby committed:
Prevent the "unknown" request type from being offered as an option on the SAR case creation page

Polsonby committed:
Data migration that sets a default request_method for SAR and Offender SAR cases

Run using `rake data:migrate` once this appllication version has been deployed in each namespace.

This should not be run for historical SAR IR cases as they don't validate this value

Polsonby committed:
Fixup spurious "puts"

Polsonby committed:
Remove the request methods helper

Polsonby committed:
Add request methods for display to decorator

Polsonby committed:
Use the new decorator for displaying request method options

Polsonby committed:
Only reference the correspondence types for which request method was added in this pull request

Polsonby committed:
Fix specs for admin case creation

Polsonby committed:
Move the request method field to the correct section of the page

rohanssalunkhe committed:
Merge branch 'main' into CDPT-7101-add-request-method-to-sar
